src/dymaptic.GeoBlazor.Core/Model/AuthenticationManager.cs

```diff
@@ -0,0 +1,187 @@
+    {
+        _jsRuntime = jsRuntime;
+        _configuration = configuration;
+    }
+
+    /// <summary>
+    ///     Initializes authentication based on either an OAuth App ID or an API Key. This is called automatically by <see cref="MapView"/> on first render, but can also be called manually for other actions such as rest calls.
+    /// </summary>
+    public async Task Initialize()
+    {
+        if (_module is null)
+        {
+            IJSObjectReference arcGisJsInterop = await GetArcGisJsInterop();
+
+            _module = await arcGisJsInterop.InvokeAsync<IJSObjectReference>("getIdentityManager",
+                _cancellationTokenSource.Token, DotNetObjectReference.Create(this), ApiKey, AppId, PortalUrl);
+        }
+    }
+
+    /// <summary>
+    ///     Initiates an OAuth login flow
+    /// </summary>
+    public async Task Login()
+    {
+        await Initialize();
+        await _module!.InvokeVoidAsync("doLogin");
+    }
+    
+    /// <summary>
+    ///     Provides an implementation-agnostic way to fetch the current authentication token, whether it's an OAuth token or an API Key
+    /// </summary>
+    public async Task<string?> GetCurrentToken()
+    {
+        if (!string.IsNullOrWhiteSpace(ApiKey))
+        {
+            return ApiKey;
+        }
+        await Initialize();
+        return await _module!.InvokeAsync<string?>("getToken");
+    }
+    
+    /// <summary>
+    ///     Tests to see if the user is logged in. True if yes, false if otherwise.
+    /// </summary>
+    /// <returns>
+    ///     Returns a boolean value indicating whether or not the user is logged in.
+    /// </returns>
+    public async Task<bool> IsLoggedIn()
+    {
+        if (!string.IsNullOrWhiteSpace(ApiKey)) return true;
+        await Initialize();
+        return await _module!.InvokeAsync<bool>("isLoggedIn");
+    }
+    
+    /// <summary>
+    ///     Ensures that the user is logged in. If not, it will attempt to log them in.
+    /// </summary>
+    public async Task EnsureLoggedIn()
+    {
+        if (!await IsLoggedIn())
+        {
+            await Login();
+        }
+    }
+    
+    /// <summary>
+    ///     Get or set the OAuth App ID.
+    /// </summary>
+    public string? AppId
+    {
+        get
+        {
+            if (string.IsNullOrWhiteSpace(_appId))
+            {
+                _appId = _configuration["ArcGISAppId"];
+            }
+
+            return _appId;
+        }
+        set
+        {
+            if (!string.IsNullOrWhiteSpace(value))
+            {
+                _appId = value;
+            }
+        }
+    }
+
+    /// <summary>
+    ///     The ArcGIS Enterprise Portal URL, only required if using Enterprise authentication.
+    /// </summary>
+    public string? PortalUrl
+    {
+        get
+        {
+            if (string.IsNullOrWhiteSpace(_portalUrl))
+            {
+                _portalUrl = _configuration["ArcGISPortalUrl"];
```

Comment:
Do we have a default set for this somewhere? What I mean is I know we don't need it if it's just arcgis online, but maybe we should still have it set so that it's consistent?

Reply:
That might be consistent for every use of `AuthenticationManager`, but across ArcGIS JS and GeoBlazor, it is more common to leave optional parameters as `null`.